res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -2413,8 +2423,16 @@ class S4Mk3Deck extends Deck {
                 this.touched = touched;
                 if (this.deck.wheelMode === wheelModes.vinyl || this.deck.wheelMode === wheelModes.motor) {
                     if (touched) {
-                        engine.setValue(this.group, "scratch2_enable", true);
-                    } else {
+                        if (useEngineScratch) {
+                            engine.scratchEnable(this.deck.currentDeckNumber,
+                                ScratchIntervalsPerRev,
+                                BaseRevolutionsPerMinute,
+                                ScratchAlpha,
+                                ScratchBeta); // ramoing true by default
+                        } else {
+                            engine.setValue(this.group, "scratch2_enable", true);
+                        }
+                    } else { // release
```

Comment:
This looks wrong - seems like `engine.setValue(this.group, "scratch2_enable", true);` is always run both in `if` and `else` if `useEngineScratch` isn't used

Reply:
I don't understand, sorry. The diff looks good, I just added an alternative path for every enable/disable branch.
But indeed, with `engine.scratchEnable()` `scratch2_enable` is also set true (in c++).

Just give a spin (haha!) when you find some time and let me know how it works for you.